File: KhoaBanking/src/main/java/com/khoa/service/TaiKhoanDangNhapService.java
package com.khoa.service;

import java.util.List;

import com.khoa.dto.ConfirmInfoDTO;
import com.khoa.dto.TaiKhoanDangNhapDTO;
import com.khoa.entity.TaiKhoanDangNhap;

public interface TaiKhoanDangNhapService {

	List<TaiKhoanDangNhapDTO> findAll();
	TaiKhoanDangNhapDTO findById(int id);
	void add(TaiKhoanDangNhapDTO dto);
	void update(TaiKhoanDangNhapDTO dto);
	void delete(int id);
	TaiKhoanDangNhapDTO findByEmailOrSodienthoai(String email, String soDienThoai);
	List<TaiKhoanDangNhapDTO> findAllDetailsTaiKhoanDangNhap();
	TaiKhoanDangNhapDTO sendEmailChangePassWord(String email, String passWord);
	TaiKhoanDangNhapDTO sendEmailChuyenTien(String email);
	int sendEmailConfirmInfo(TaiKhoanDangNhapDTO taiKhoanDangNhapDTO);
	TaiKhoanDangNhapDTO confirmInfo(ConfirmInfoDTO confirmInfoDTO);
	int createEmployee(TaiKhoanDangNhap taiKhoanDangNhap);
	List<TaiKhoanDangNhap> findAllEmployee();
	List<TaiKhoanDangNhap> searchTaiKhoanNhanVien(String email);
	TaiKhoanDangNhap updateEmployee(TaiKhoanDangNhap taiKhoanDangNhap);
	TaiKhoanDangNhap deleteEmployee(int mataikhoan);
	TaiKhoanDangNhapDTO sendEmailCheckAccountExisted(String email);
}

File: internetbanking/src/app/services/taikhoandangnhap.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from 'src/environments/environment';

@Injectable({
    providedIn: 'root'
})
export class TaiKhoanDangNhapService {
    constructor(private http: HttpClient) { }

    findAllDetailsTaiKhoanDangNhap(value: any) {
        return this.http.post(`${environment.apiUrl}/taikhoanthanhtoan/findalltaikhoanthanhtoan`, value);
    }

    sendEmailChangePassWord(value: any) {
        return this.http.post(`${environment.apiUrl}/taikhoandangnhap/sendEmailChangePassWord`, value);
    }


    confirmChangePassWord(value: any) {
        return this.http.post(`${environment.apiUrl}/taikhoandangnhap/confirmChangePassWord`, value);
    }

    sendEmailChuyenTien(value: any) {
        return this.http.post(`${environment.apiUrl}/taikhoandangnhap/sendEmailChuyenTien`, value);
    }

    sendEmailConfirmInfo(value: any) {
        return this.http.post(`${environment.apiUrl}/taikhoandangnhap/sendEmailConfirmInfo`, value);
    }

    confirmInfo(value: any) {
        return this.http.post(`${environment.apiUrl}/taikhoandangnhap/confirmInfo`, value);
    }

    createEmployee(value: any) {
        return this.http.post(`${environment.apiUrl}/taikhoandangnhap/createEmployee`, value);
    }

    findAllEmployee() {
        return this.http.get(`${environment.apiUrl}/taikhoandangnhap/findAllEmployee`);
    }

    updateEmployee(value: any) {
        return this.http.post(`${environment.apiUrl}/taikhoandangnhap/updateEmployee`, value);
    }

    deleteEmployee(value: { mataikhoan: any; }) {
        return this.http.post(`${environment.apiUrl}/taikhoandangnhap/deleteEmployee`, value);
    }
}